### ISSUE TITLE
Syntax error with latest jQuery

### ISSUE THREAD

**Issue by matatk** [jmpressjs/jmpress.js#185]
Hi and thank you for this brilliant tool.  I just started using it, and noticed that if I am using jQuery 2.2.1 then I get a syntax error:

    Syntax error, unrecognized expression: a[href^=#]

However if I use jQuery 2.1.4, as you do, the error does not occur.

I think it is to do with [the need to quote special characters in CSS selectors](http://stackoverflow.com/questions/31197452/syntax-error-unrecognized-expression-for-href).

**Comment by FagnerMartinsBrack** [comment]
Could you post a [jsfiddle](https://jsfiddle.net) or [jsbin](http://jsbin.com/?html,output) that reproduces the issue?

**Comment by matatk** [comment]
Sure, this demonstrates the problem: http://jsbin.com/qozolo/edit?html,output

If you switch from jQuery 2.2.1 to 2.1.4 then the syntax error doesn't happen.

**Comment by FagnerMartinsBrack** [comment]
You are using jmpress.js. This project is not jmpress, this is impress.js and don't use jQuery. You might want to report this on https://github.com/jmpressjs/jmpress.js.

**Comment by FagnerMartinsBrack** [comment]
Thanks, it is much easier for the authors of the project fix the issue if you provide an example :)

**Comment by shama** [comment]
@FagnerMartinsBrack Now realizing that I don't have the time I thought I would for this project and you are actively helping, would you be interested in being added as a contributor?

**Comment by FagnerMartinsBrack** [comment]
@shama Thanks for the proposal.

Recently I have been added as a contributor in the original [impress.js project](https://github.com/impress/impress.js) in order to shift the code from a presentation only framework to a core-only, plugin-based library ([see this comment](https://github.com/impress/impress.js/issues/435#issuecomment-177602861)). One of the reasons is that a have a project that depends on impress.js working like only the core, not as a full coupled presentation framework. Since there is a lot of issues/PR I expect that the end-goal will take many months because I am not as active in open-source as many ppl out there.

After impress.js becomes more decoupled, I see a very interesting use-case for jmpress.js. jmpress.js would be more about **running an actual talk presentation on infinite canvas (in a website or slide projector)** than provide only a **decoupled core** for extension, so it can consume impress.js in the future and reduce the amount of code that needs to be tested. Also, it might not need jQuery anymore in the future since the heavy stuff will be done by impress.js.

I would say that I have more stake in a project that contains only a simple, lightweight, core presentation lib than one with many features for specific use cases.

TL;DR for the future:

* impress.js -> 3D presentation core
* jmpress.js -> 3D presentation specific for Talks in the web or slide projector (maybe then rename to avoid confusion?)

If you agree with the goals above, you can add me as a contributor. But I don't expect to work in jmpress.js anytime soon in this year, other than triaging a few issues/PRs here and there until impress.js is ready.

**Comment by hroma** [comment]
Quick fix for jQuery 22.1.
Find
`hashLink = "a[href^=#]";`
and replace with
`hashLink = 'a[href^="#"]';`

Tested with jQuery 2.2.1. Result = syntax error is gone.

**Comment by FagnerMartinsBrack** [comment]
The selector `a[href^=#]` is invalid and jQuery fixed it recently, that's why it is breaking. See https://github.com/jquery/jquery/issues/2824
